src/components/MapComponent.tsx

```diff
@@ -70,136 +69,101 @@ export const MapComponent: React.FC<Props> = ({ options, width, height, data, on
       ...object,
       positions: object.positions.filter((position) => isSameDay(position.timestamp, new Date(dateToDisplay))),
     }))
-    .filter((object) => object.positions.length > 0);
+    .filter((object) => object.positions.length > 0)
+    .filter(object => selectedObjectId === 'all' ? object : String(object.id) === selectedObjectId
```

Comment:
This condition is strange. Shouldn't it be object => object.id === selectedObjectId || selectedObjectId==='all'?